CMS-Enterprise/easi-app
# [EASI-4282] System intake form refactor

# EASI-4282


## Changes and Description

- Refactored System Intake form Request Details and Contract Details sections to use React Hook Forms
  - Contact Details refactor was completed in previous PR
- Fixed issue with auto save in Contact Details section of form
- Refactored Funding Sources component with RHF
  - Created new component (/components/FundingSources) for system intake form
  - Moved old FundingSources component to TRB folder



## How to test this change
- Test system intake form
- Make sure form auto saves changes
- "Back" and "save and exit" buttons should save data but not validate fields



## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
